test/e2e/density.go

```diff
@@ -187,8 +187,14 @@ var _ = PDescribe("Density", func() {
 	})
 
 	AfterEach(func() {
-		// Remove any remaining pods from this test
-		DeleteRC(c, ns, RCName)
+		// Remove any remaining pods from this test if the
+		// replication controller still exists and the replica count
+		// isn't 0.  This means the controller wasn't cleaned up
+		// during the test so clean it up here
+		rc, err := c.ReplicationControllers(ns).Get(RCName)
+		if err == nil && rc.Spec.Replicas != 0 {
```

Comment:
If there's a problem communicating with the apiserver then there would likely be an rc and its pods still in the system.  That could definitely impact other tests.  I'm not sure there's much we can do though.  If the apiserver is unresponsive I'm not sure how we clean up after the test.